app/forms/providers/means/regular_income_form.rb

```diff
@@ -105,33 +105,35 @@ def destroy_legal_aid_application_transaction_types!
         legal_aid_application
           .legal_aid_application_transaction_types
           .credits
-          .where.not(transaction_type_id: transaction_type_ids)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
       def destroy_regular_income_transactions!
         legal_aid_application
           .regular_incomes
-          .without(existing_income_regular_transactions)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
       def destroy_cash_income_transactions!
         legal_aid_application
           .cash_transactions
           .credits
-          .where.not(transaction_type_id: transaction_type_ids)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
-      def existing_income_regular_transactions
-        legal_aid_application
-          .regular_incomes
-          .where(transaction_type_id: transaction_type_ids)
+      def transaction_type_ids_to_keep
+        transaction_type_ids.append(housing_benefit_id)
```

Comment:
I don't think we should be mutating the `transaction_type_ids` attribute in this way.

This is probably leading to unintended side-effects when it comes to associating transaction types to the application.

Reply:
agreed to rip this out in next optimisation PR

Reply:
done